File: atcoder/abc071C_make_a_rectangle.py
# Vicfred
# https://atcoder.jp/contests/abc071/tasks/arc081_a
# greedy, sorting
class Counter(dict):
    def __missing__(self, i):
        return 0

n = int(input())
a = list(map(int, input().split()))

a.sort()
a.reverse()

counter = Counter()
for i in a:
    counter[i] += 1

first = -1
second = -1
for item in a:
    if second != -1:
        break;
    if counter[item] >= 2:
        if first == -1:
            first = item
            counter[item] -= 2
        elif second == -1:
            second = item

if first != -1 and second != -1:
    print(first*second)
else:
    print(0)

File: atcoder/abc071C_make_a_rectangle.nim
# unihernandez22
# https://atcoder.jp/contests/abc071/tasks/arc081_a
# implementation

from algorithm import sort, reversed
from strutils import split, parseInt
from sequtils import map
import tables

discard stdin.readLine
var a = stdin.readLine.split.map parseInt

a.sort cmp
var b = initTable[int, int]()

for i in a:
  b[i] = b.getOrDefault(i) + 1

var height: int
var width: int
for i in a.reversed:
  if b[i] >= 2:
    if  height == 0: height = i
    elif width == 0: width = i
    b[i] -= 2

echo width * height

